### ISSUE TITLE
[ENH] support for multi-column stratification and leave-one-group-out validation

### ISSUE THREAD

**Issue by kaurao** [juaml/julearn#118]
Two sampling strategies

- multi-column stratification https://stackoverflow.com/questions/45516424/sklearn-train-test-split-on-pandas-stratify-by-multiple-columns/
- leave-one-group-out sampling defined by one column (e.g  site) where each unique value will become the test set (number of unique values = number of folds)

**Comment by fraimondo** [comment]
Any use case? example?

**Comment by fraimondo** [comment]
closed due to inaction
